### ISSUE TITLE
negative dimensions error in groopm core

### ISSUE THREAD

**Issue by koadman** [Ecogenomics/GroopM#13]
Howdy gents, I have stumbled over the following using groopm core (0.3.4 from pip).

**On stderr**:
  File "/shared/homes/117335/bin/groopm", line 381, in <module>
    GM_parser.parseOptions(args)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/groopm.py", line 138, in parseOptions
    gf=gf)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 208, in makeCores
    self.initialiseCores(kmerThreshold, coverageThreshold)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 247, in initialiseCores
    putative_clusters = self.findNewClusterCenters(kmerThreshold, coverageThreshold)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 405, in findNewClusterCenters
    coverageThreshold)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 670, in twoWayContraction
    (k_partitions, k_keeps) = self.HP.houghPartition(k_dat[:,0], l_dat)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 1360, in houghPartition
    imgTag=imgTag)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 1503, in recursiveSelect
    (m, ret_point, accumulator) = self.houghTransform(tData.astype(float)[startRange:endRange,:], imShape)
  File "/shared/homes/117335/lib/python2.7/site-packages/groopm/cluster.py", line 1730, in houghTransform
    Cs = np_array(range(cols)*d_len)
ValueError: negative dimensions are not allowed

**Standard output has the following**:

*******************************************************************************
 [[GroopM 0.3.4]] Running in core creation mode...
*******************************************************************************
Loading data from: groopdb
    GroopM DB version (5) up to date
    Loaded indices with condition: length >= 1500
    Working with: 597943 contigs
    Loading coverage profiles
    Loading PCA kmer sigs (42 dimensional space)
    Loading PCA kmer variance (total variance: 0.80)
    Loading contig names
    Loading contig lengths (Total: 1831237672 BP)
    Loading contig GC ratios (Average GC: 0.408)
    Creating color map
    { THIS: 0:47:35.123 || TOTAL: 0:47:35.165 }
    Loading transformed data
    Reticulating splines
    { THIS: 0:06:47.138 || TOTAL: 0:54:22.303 }
Create cores
     .... .... .... .... .... .... .... .... .... ....
   0 Unexpected error: <type 'exceptions.ValueError'>

**I ran**: 
groopm core -b 500000 groopdb

The only possibly unusual issues are that (1) many contigs have no mapped reads, and (2) two earlier attempts at groopm core on the same DB were mercilessly killed by the job scheduler for exceeding RAM limits.

**Comment by wwood** [comment]
Hmm, thanks for the report. Any chance you can share the raw data privately so we can investigate with less guesswork please?

**Comment by koadman** [comment]
Possibly, lemme check with the person who gave it to me.
If so, how to send you 107Gb of .bam files?

**Comment by wwood** [comment]
OK, would sending the groopm dB be smaller?

**Comment by koadman** [comment]
good idea, that's only 12Gb. I'll send a CloudStor link your way when I can get it uploaded from campus

**Comment by wwood** [comment]
Hey @koadman we've (i.e. Tim) has managed to track at least part of this bug to what seems to be a bug in sciypy where creation from lists >2^13-1 in length doesn't work as expected. Still getting a separate error now though, Tim continues..

**Comment by koadman** [comment]
thanks for the update! darn those pesky 32 bit integers.

**Comment by pavia27** [comment]
Morning,

Would it be possible to share how you remedied this error? I have been getting the same issue (negative dimensions are not allowed), but my gm is only 3.9GB. 

Any help would be greatly appreciated!

Best, 
Mike
